spec/SPEC.md

```diff
@@ -89,259 +79,469 @@ produced the executable.
 of changes to those files across all users of those projects. By leveraging
 transparent inclusion of Input Manifests into executable and other formats,
-users would also gain the benefits of high precision supply chain information
+users would also gain the benefits of high precision dependency information
 without manually recording or updating those manifests as projects develop over
 time.
 
-##	Scope
+## 3. Scope
 
-Specifies procedures for constructing and conveying Input Manifests,
-Artifact Dependency Graphs (ADGs), and other related data structures 
-for artifacts. Including but not limited to:
+This document specifies:
 
-- formats for artifact identifiers
-- formats for specifying graph relationships between artifacts
-- manner of embedding identifiers for Input Manifests, ADGs, and other related
-  data structures in artifacts of various types
-- guidance on metadata which references Input Manifests, ADGs, and other related data structures
-- guidance for build tools for:
-  - constructing Input Manifests, ADGs, and other related data structures
-  - conveying Input Manifests, ADGs, and other related data structures
-  - embedding identifiers for Input Manifests, and other related data structures' ids in artifacts
-  - manners of conveyance of Input Manifests, and other related data structure's
-  - descriptions of use cases for which Input Manifests, ADGs, and other related data structures may be used
+- The in-memory format of Artifact Identifiers
+- The textual representation of Artifact Identifiers
+- The process for constructing Artifact Identifiers
+- The textual representation of Input Manifests
+- How Input Manifests should be stored in a file system
+- How Input Manifests should be embedded into the artifacts whose build
+  inputs they are describing
+- How Input Manifests should be constructed by build tools.
 
-##	Normative References
+##	4. Normative References
 
 - [GitOID URI][gitoid_uri]
 
-## Terms and Definitions
+## 5. Terms and Definitions
 
 For the purposes of this document, the following terms and definitions apply.
 
-### Artifact
+### 5.1. Artifact
 
-An artifact is any object of interest that can be represented as arrays of
-bytes (`[]byte`).
+An artifact is any object of interest that can be represented as an array of
+bytes.
 
-### Artifact Equivalency
+### 5.2. Artifact Equivalency
 
 Two artifacts are equivalent if and only if their binary representations are
-equal. This can be expressed in pseudocode with the following expression:
-`[]byte(artifact1) == []byte(artifact2)`
+equal.
+
+### 5.3. Build Tools
+
+A build tool is anything which reads one or more input artifacts and writes
+one or more output artifacts. Examples of build tools include:
+
+- __Compilers:__
+  - `llvm-clang`
+  - `gcc`
+  - `javac`
+  - `rustc`
+  - `go`
+- __Linkers:__
+  - `llvm-lld`
+  - `binutils-ld`
+- __Runtimes__
+  - Java JVM
+  - Node.js
+  - Python interpreter
+- __Code Generators__
+
```

Comment:
Could we work patch into here?

Reply:
Looks like the use of colons is also inconsistent, I ought to fix that.